### DISCUSSION TITLE
How Can We Make PR Review More Efficient?

### DISCUSSION THREAD

**Discussion by nilsnolde** [valhalla/valhalla#3310] [General]
There's been quite a lot of activity recently by the community and (at least speaking for myself as part of "community") that'll likely only gonna increase in the near future. However, I feel like the reviewing burden could be a little more distributed, it's mostly @kevinkreiser reviewing almost _every_ PR (not paid!). 

I think there's quite a few companies (probably more than I know of) using Valhalla successfully in their projects, proprietary or not. And I'm sure some of those companies have one or the other c++ dev who could potentially contribute to reviewing. That's where IMHO most of the burden should be, as we (as in we companies using Valhalla to generate revenue) get an amazing project for no $ and the only currency we can use to pay back is dev time. Alternatively we could set up Github Sponsors and instead of putting in dev time companies or others could donate. How to potentially distribute that sponsorship cash is a different topic though. Maybe there's other good options?

I'd love to be able to review more in-depth myself, but mostly I don't feel comfortable reviewing complex c++ PRs (yet). Working on that.. I'd pledge to brush up my c++ fu to step up with reviewing a little more. Who's with me?

**Comment by kevinkreiser** [comment]
An idea hit me this morning. Here and there I will pair up with people to review their code directly (via video call) rather than try to review it asynchronously. In this way we can vastly speed up the review process because the person who wrote the code is right there to answer questions right at that point in time. What if rather than relying on me to facillitate that in an adhoc fashion, we instated a recurring video call, where anyone with an open PR can show up and talk through the review?

I'm thinking that once a week we could spend an hour reviewing PRs of anyone who would like to join. I know its a global project so finding a good time that meets everyones schedules will be hard to do. So maybe the week prior we can have people put themselves on the coming weeks agenda and before the meeting happens we can hone in on a time of day that would work for them and for the potential reviewer. That way we both know what will be discussed and when, in advance of it. Does anyone else have any ideas?

**Reply by nilsnolde** [reply]
big +1 of course :) I can open and maintain a document where people can just put their name/PR/tz every week. we can reference that doc in the contributions part of the README?

**Comment by nilsnolde** [comment]
smth like that @kevinkreiser ? https://docs.google.com/spreadsheets/d/19TrKmdjWu0FVOlF3-cYHAZOFXLg8zI6_GHWPzWTVyIc/edit?usp=sharing. should we put a link to the contributing section in the readme?

I'd keep it simple, one sheet per week, people sign in with their email and on fridays before I send out some doodle or so to find a common time. I happily maintain that document and the process

**Reply by kevinkreiser** [reply]
one thing i was thinking was that maybe we could start a new discussion at the beginning of each week asking for a roll call for the next weeks PR review. i like github discussions better than google sheets because we dont have to worry about random people editing the document. although i agree the spreadsheet is easier to read quickly than a discussion thread. maybe ive overlooked something about  google sheets though so feel free to englighten me :smile:

**Reply by kevinkreiser** [reply]
i suppose you could use google sheets comments to avoid people messing with other peoples edits, that could work

**Reply by nilsnolde** [reply]
haha maybe I'm too hippie, I'd just trust people to not mess with other people's stuff (I don't like any sheets shite really, but seemed like good enough for this). didn't want to overload discussions every week and a single discussion for all weeks seems too much info as well IMO. 

should we try with google sheets editor access and if someone complains we do commenting?

hm just reminded of QGIS open day, there they use their wiki where everyone can edit ([github setting](https://docs.github.com/en/communities/documenting-your-project-with-wikis/changing-access-permissions-for-wikis)). would mean we'd have to open the entire wiki for everyone but the good thing is it's accountable with a history and changesets. 

how's that?

**Reply by nilsnolde** [reply]
an example page from qgis: https://github.com/qgis/QGIS/wiki/QHF-March-2021

we could simply rename every week and nuke the tables in there after each meeting

**Reply by nilsnolde** [reply]
smth like that as a wiki page? 

# Valhalla Open Review Days

Once a week we organize an hour of discussing open PRs to speed up the review process. People are welcome to join, priority will be PR reviews, but if time allows general questions/feedback are also much welcome.

Below you'll find a list where you can register with your name, email and preferred days & times (all in UTC please). At the end of each week we'll determine the exact day and time based on the registrants and send out an invitation to a virtual meeting.

### Registration for week 04.10. - 10.10.2021 (week 40)

If you want to participate, please edit this document and add your details to an empty row. Your email is optional and will be used to announce the final day & time.

| Name       	| Email            	| PR/Topic                                                	| Timezone 	| Preferred day/time (in UTC) 	|
|------------	|------------------	|---------------------------------------------------------	|----------	|-----------------------------	|
| Nils Nolde 	| nils at gis-ops dot com 	| [#3206](https://github.com/valhalla/valhalla/pull/3206) 	| Berlin   	| Mo - Fr, 7 am - 3 pm        	|
|            	|                  	|                                                         	|          	|                             	|
|            	|                  	|                                                         	|          	|                             	|
|            	|                  	|                                                         	|          	|                             	|

**Reply by kevinkreiser** [reply]
done

https://github.com/valhalla/valhalla/wiki/Open-Review-Days
